parser/lexer.go

```diff
@@ -443,6 +443,12 @@ func startWithNumber(s *Scanner) (tok int, pos Pos, lit string) {
 	if ch0 == '.' || ch0 == 'e' || ch0 == 'E' {
 		return s.scanFloat(&pos)
 	}
+
+	// Identifiers may begin with a digit but unless quoted may not consist solely of digits.
```

Comment:
Why check "ch0 != unicode.ReplacementChar"?

Any example of "unless quoted may not consist solely of digits"?